utils/python/CIME/create_test.py

```diff
@@ -105,15 +114,17 @@ def __init__(self, test_names,
         # This is the only data that multiple threads will simultaneously access
         # Each test has it's own index and setting/retrieving items from a list
         # is atomic, so this should be fine to use without mutex
-        self._test_states = [ (INITIAL_PHASE, TEST_PASS_STATUS) ] * len(self._test_names)
+        for test in self._tests:
```

Comment:
I've read this a few times and I think I understand your concern, but I do not see the danger of which you speak.  Can you give me a concrete example of what you think could go wrong?